vsphere_plugin_common/clients/server.py

```diff
@@ -756,42 +760,53 @@ def create_server(
                 server=vm_name, template=template_name))
         self._logger.debug('Cloning with clonespec: {spec}'
                            .format(spec=text_type(clonespec)))
-        task = template_vm.obj.Clone(folder=destfolder,
-                                     name=vm_name,
-                                     spec=clonespec)
         try:
-            self._logger.debug(
-                "Task info: {task}".format(task=text_type(task)))
-            # wait for task finish
-            self._wait_for_task(task,
-                                max_wait_time=max_wait_time,
-                                resource_id=VSPHERE_SERVER_ID)
+            if not retry:
+                task = template_vm.obj.Clone(folder=destfolder,
+                                             name=vm_name,
+                                             spec=clonespec)
+                self._logger.debug(
+                    "Task info: {task}".format(task=text_type(task)))
+                # wait for task finish
+                self._wait_for_task(task,
+                                    max_wait_time=max_wait_time,
+                                    resource_id=VSPHERE_SERVER_ID)
+            else:
+                self._wait_for_task(max_wait_time=max_wait_time,
+                                    resource_id=VSPHERE_SERVER_ID)
 
             ctx.instance.runtime_properties['name'] = vm_name
             ctx.instance.runtime_properties.dirty = True
             ctx.instance.update()
 
-            if enable_start_vm:
+            if not retry and enable_start_vm:
                 self._logger.info('VM created in running state')
                 while not self._wait_vm_running(
                     task.info.result, adaptermaps, os_type == "other"
                 ):
                     time.sleep(TASK_CHECK_SLEEP)
             else:
```

Comment:
@EarthmanT in retry it will be here so , I think enclose the whole if and else with not retry

Reply:
I guess that is not a big deal since an exception of OperationRetry will be thrown before it reaches here , so I guess no need